### ISSUE TITLE
Cannot install package vector

### ISSUE THREAD

**Issue by kranich** [valderman/haste-compiler#222]
The installation of package `vector` fails due to its dependence on package `primitive` (see also #190).

I run the latest haste from git with ghc 7.6.3 on linux.

`haste-inst install primitive` produces the following log:
```
Building primitive-0.5.3.0...
Preprocessing library primitive-0.5.3.0...

Data/Primitive/Types.hs:153:268:
    Couldn't match expected type `Word64#' with actual type `Word#'
    In the return type of a call of `indexWord64Array#'
    In the first argument of `W64#', namely
      `(indexWord64Array# arr# i#)'
    In the expression: W64# (indexWord64Array# arr# i#)

Data/Primitive/Types.hs:153:441:
    Couldn't match expected type `Word64#' with actual type `Word#'
    In the first argument of `W64#', namely `x#'
    In the expression: W64# x#
    In the expression: (# s1#, W64# x# #)

Data/Primitive/Types.hs:153:519:
    Couldn't match expected type `Word#' with actual type `Word64#'
    In the third argument of `writeWord64Array#', namely `x#'
    In the expression: writeWord64Array# arr# i# x# s#
    In an equation for `writeByteArray#':
        writeByteArray# arr# i# (W64# x#) s#
          = writeWord64Array# arr# i# x# s#

Data/Primitive/Types.hs:153:843:
    Couldn't match expected type `Word64#' with actual type `Word#'
    In the return type of a call of `indexWord64OffAddr#'
    In the first argument of `W64#', namely
      `(indexWord64OffAddr# addr# i#)'
    In the expression: W64# (indexWord64OffAddr# addr# i#)

Data/Primitive/Types.hs:153:1019:
    Couldn't match expected type `Word64#' with actual type `Word#'
    In the first argument of `W64#', namely `x#'
    In the expression: W64# x#
    In the expression: (# s1#, W64# x# #)

Data/Primitive/Types.hs:153:1099:
    Couldn't match expected type `Word#' with actual type `Word64#'
    In the third argument of `writeWord64OffAddr#', namely `x#'
    In the expression: writeWord64OffAddr# addr# i# x# s#
    In an equation for `writeOffAddr#':
        writeOffAddr# addr# i# (W64# x#) s#
          = writeWord64OffAddr# addr# i# x# s#

Data/Primitive/Types.hs:168:265:
    Couldn't match expected type `Int64#' with actual type `Int#'
    In the return type of a call of `indexInt64Array#'
    In the first argument of `I64#', namely
      `(indexInt64Array# arr# i#)'
    In the expression: I64# (indexInt64Array# arr# i#)

Data/Primitive/Types.hs:168:436:
    Couldn't match expected type `Int64#' with actual type `Int#'
    In the first argument of `I64#', namely `x#'
    In the expression: I64# x#
    In the expression: (# s1#, I64# x# #)

Data/Primitive/Types.hs:168:513:
    Couldn't match expected type `Int#' with actual type `Int64#'
    In the third argument of `writeInt64Array#', namely `x#'
    In the expression: writeInt64Array# arr# i# x# s#
    In an equation for `writeByteArray#':
        writeByteArray# arr# i# (I64# x#) s#
          = writeInt64Array# arr# i# x# s#

Data/Primitive/Types.hs:168:836:
    Couldn't match expected type `Int64#' with actual type `Int#'
    In the return type of a call of `indexInt64OffAddr#'
    In the first argument of `I64#', namely
      `(indexInt64OffAddr# addr# i#)'
    In the expression: I64# (indexInt64OffAddr# addr# i#)

Data/Primitive/Types.hs:168:1010:
    Couldn't match expected type `Int64#' with actual type `Int#'
    In the first argument of `I64#', namely `x#'
    In the expression: I64# x#
    In the expression: (# s1#, I64# x# #)

Data/Primitive/Types.hs:168:1089:
    Couldn't match expected type `Int#' with actual type `Int64#'
    In the third argument of `writeInt64OffAddr#', namely `x#'
    In the expression: writeInt64OffAddr# addr# i# x# s#
    In an equation for `writeOffAddr#':
        writeOffAddr# addr# i# (I64# x#) s#
          = writeInt64OffAddr# addr# i# x# s#
```

**Comment by vzaccaria** [comment]
Same problem here installing package `matrix`. 

Looks like `primitive` uses signatures from `GHC.Prim` which, in turn, provides non compliant signatures (e.g., for `eqAddr#`):

```
Data/Primitive/Types.hs:50:33:
    Couldn't match expected type ‘Bool’ with actual type ‘Int#’
    In the first argument of ‘isTrue#’, namely ‘(eqAddr# a# b#)’
    In the expression: isTrue# (eqAddr# a# b#)
```

**Comment by valderman** [comment]
Fixed on master with GHC 7.10.
